ui-participant/src/hub/survey/SurveyView.test.tsx

```diff
@@ -230,16 +244,25 @@ describe('Renders a survey', () => {
 const setupSurveyTest = (survey: Survey) => {
   const submitSpy = jest.spyOn(Api, 'updateSurveyResponse')
     .mockImplementation(() => Promise.resolve(mockHubResponse()))
+  const autosaveManager = {
+    trigger: (): void => { throw 'no autosave registered' }
+  };
+
+  (useAutosaveEffect as jest.Mock).mockImplementation(saveFn => {
+    autosaveManager.trigger = () => { saveFn() }
```

Comment:
Trying to connect the mocking dots... the `saveFn` here is a no-op function assigned at line 22?

Reply:
the saveFn is what will be passed into the useAutosaveEffect by downstream consumers.  So, e.g. SurveyView will pass in the function that calls the save response API.